src/analyze/analyze.c

```diff
@@ -458,21 +467,21 @@ static int acquire_host_info(sd_bus *bus, struct host_info **hi) {
                                    hostname_map,
                                    host);
         if (r < 0)
-                goto fail;
+                log_debug("Failed to get host information from systemd-hostnamed");
 
         r = bus_map_all_properties(bus,
                                    "org.freedesktop.systemd1",
                                    "/org/freedesktop/systemd1",
                                    manager_map,
                                    host);
-        if (r < 0)
-                goto fail;
+        if (r < 0) {
+                log_error("Failed to get host information from systemd");
```

Comment:
Also, if you do `free_host_info(host);` first, you can make use of the fact that `log_error_errno()` returns its first argument again, so you can simply `return log_error_errno();`